squid: remove knownVulnerabilities for 7.0+

longregen opened the pull request:
In the past, it has been brought up that Squid had many security vulnerabilities (see https://megamansec.github.io/Squid-Security-Audit/). As of version 7.0,  all of them have been solved, as tracked in their GitHub Security page: https://github.com/squid-cache/squid/security

## Things done

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [ ] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

fpletz reviewed (changes requested):
Welcome to nixpkgs! 👋 

Please use `nixfmt` to fix the formatting issues.

fpletz commented:
Let's backport this since 25.05 also contains version 7.0.1.

longregen commented:
> Welcome to nixpkgs! 👋
> 
> Please use `nixfmt` to fix the formatting issues.

Thank you, really sorry; formatted.

> Let's backport this since 25.05 also contains version 7.0.1.

I'll follow https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#how-to-backport-pull-requests

longregen committed:
squid: remove knownVulnerabilities for 7.0+

In the past, it has been brought up that Squid had many security vulnerabilities
(see https://megamansec.github.io/Squid-Security-Audit/). As of version 7.0,
all of them have been solved, as tracked in their GitHub Security page:
https://github.com/squid-cache/squid/security

fpletz commented:
> I'll follow [`master`/CONTRIBUTING.md#how-to-backport-pull-requests](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md?rgh-link-date=2025-07-31T16%3A57%3A14Z#how-to-backport-pull-requests)

The automated workflow will take care of that. :smiley:
